### PR TITLE
CORE-1706, CORE-1707, CORE-1709 reachpc CD scripts

### PR DESCRIPTION
**Pull request by erer1243**
Release build scripts for reachpc (and reachc since it's unavoidable with a single stack project -- this might be useful if we want to have native reachd-local distributions, otherwise it's just a waste of money). 

Since reach-cli cannot be compiled for Windows, I ripped out the incompatible functionality. When we get far enough into cloud to remove reach-cli from the repo, and we can actually everything compile for windows, I will rebase this branch and we can actually merge it. For now I'm just opening this pr for review. 

Currently I have release builds require an approve action, since they take a lot of time and cost ~$2 per run. Do we want to keep this? Or have it run once a day? Or some other system?

### CODE REVIEW

**Comment by jeapostrophe** [comment]
I am writing this before reading the actual changes

> Release build scripts for reachpc (and reachc since it's unavoidable with a single stack project -- this might be useful if we want to have native reachd-local distributions, otherwise it's just a waste of money).

I thought when you build with stack, you can specify the target?

> Since reach-cli cannot be compiled for Windows, I ripped out the incompatible functionality. When we get far enough into cloud to remove reach-cli from the repo, and we can actually everything compile for windows, I will rebase this branch and we can actually merge it. For now I'm just opening this pr for review.

That makes sense

> Currently I have release builds require an approve action, since they take a lot of time and cost ~$2 per run. Do we want to keep this? Or have it run once a day? Or some other system?

I think they should be part of the approve process

File: .circleci/config.pre.yml [hunk 1]
```diff
@@ -350,6 +382,16 @@ workflows:
            only: master
        requires:
        - "docs-render"
+  release_build:
+    jobs:
+      - approve-release-builds:
+          type: approval
```
Comment (jeapostrophe) [inline .circleci/config.pre.yml:388]: There already is an approve step for doing releases

**Review by jeapostrophe — COMMENTED** [review]
You can have PRs into branches other than `master`

**Comment by erer1243** [comment]
>I thought when you build with stack, you can specify the target?
You can, but stack will build everything anyway at least once.

File: hs/stack.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@ resolver: lts-19.7
 ghc-options:
   "$everything": -haddock
 
-# system-ghc: true
+system-ghc: true
```
Comment (jeapostrophe) [inline hs/stack.yaml:6]: !

**Review by jeapostrophe — COMMENTED** [review]
The merge is messed up

**Comment by jeapostrophe** [comment]
> > I thought when you build with stack, you can specify the target?
> 
> You can, but stack will build everything anyway at least once.

That's weird and I feel like there HAS to be a solution. In the worst case, you can do some silly stuff and make it so that build script edits the package.yaml file before building it

**Comment by jeapostrophe** [comment]
Dead

**Pull request closed without merging**